api/src/main/java/ai/djl/nn/recurrent/LSTM.java

```diff
@@ -128,27 +129,28 @@ protected NDList opInputs(ParameterStore parameterStore, NDList inputs, boolean
         validateInputSize(inputs);
         long batchSize = inputs.head().getShape().get(0);
         inputs = updateInputLayoutToTNC(inputs);
-        NDArray head = inputs.singletonOrThrow();
+        NDArray head = inputs.head();
+        NDManager manager = head.getManager();
         Device device = head.getDevice();
 
         NDList result = new NDList(head);
         try (NDList parameterList = new NDList()) {
             for (Parameter parameter : parameters.values()) {
-                NDArray array = parameterStore.getValue(parameter, device, training);
+                NDArray array = parameterStore.getValue(parameter, device, training).duplicate();
```

Comment:
Why we do duplicate here? Does that cause any issues?

Reply:
Changed.